packages/matter.js/src/cluster/ActionsCluster.ts

```diff
@@ -0,0 +1,345 @@
+
+    /** This command disables a certain action or automation with a duration. */
+    disableActionWithDuration: BitFlag(11),
+
+});
+
+/**
+ * Action error.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.3
+ */
+export const enum ActionErrorEnum {
+    Unknown = 0,
+
+    Interrupted = 1,
+
+}
+
+/**
+ * Action state.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.3
+ */
+export const enum ActionStateEnum {
+    /** The action is not active */
+    Inactive = 0,
+
+    /** The action is active */
+    Active = 1,
+
+    /** The action has been paused */
+    Paused = 2,
+
+    /** The action has been disabled */
+    Disabled = 3,
+}
+
+/**
+ * Types of actions.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.2
+ */
+export const enum ActionTypeEnum {
+    /** Use this only when none of the other values applies */
+    Other = 0,
+
+    /** Bring the endpoints into a certain state */
+    Scene = 1,
+
+    /** A sequence of states with a certain time pattern */
+    Sequence = 2,
+
+    /** Control an automation (e.g. motion sensor controlling lights) */
+    Automation = 3,
+
+    /** Sequence that will run when something doesn’t happen*/
+    Exception = 4,
+
+    /** Use the endpoints to send a message to user*/
+    Notification = 5,
+
+    /** Higher priority notification */
+    Alarm = 6,
+}
+
+/**
+ * Provides information about endpoint conformance to a release of a device type definition.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.6
+ */
+const TlvAction = TlvObject({
+    /** Indicates the device type definition */
+    actionID: TlvField(0, TlvUInt16),
+
+    /** Indicates the implemented revision of the device type definition */
+    name: TlvField(1, TlvString.bound({ maxLength: 32 })),
+
+    type: TlvField(2, TlvEnum<ActionTypeEnum>()),
+
+    endpointListID: TlvField(3, TlvUInt16),
+
+    supportedCommands: TlvField(4, CommandBits),
+
+    state: TlvField(5, TlvEnum<ActionStateEnum>()),
+
+});
+
+/**
+ * Provides information about endpoint conformance to a release of a device type definition.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.6
+ */
+export const enum EndpointListType {
+    /** Another group of endpoints */
+    Other = 0,
+
+    /** User-configured group of endpoints where an endpoint can be in only one room */
+    Room = 1,
+
+    /** user-configured group of endpoints where an endpoint can be in any number of zones */
+    Zone = 2,
+}
+
+/**
+ * Provides information about endpoint conformance to a release of a device type definition.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.5
+ */
+const TlvEndpointList = TlvObject({
+    /** Indicates the device type definition */
+    endpointListId: TlvField(0, TlvUInt16),
+
+    /** Indicates the implemented revision of the device type definition */
+    name: TlvField(1, TlvString),
+
+    type: TlvField(2, TlvEnum<EndpointListType>()),
+
+    endpoints: TlvField(3, TlvArray(TlvUInt16)),
```

Comment:
```suggestion
    /** List of endpoint numbers. */
    endpoints: TlvField(3, TlvArray(EndpointNumber)),
```

(and add import please!)

Reply:
and maxLength: 256

Reply:
Done.